ckanext/unaids/react/components/FileInputComponent/src/App.js

```diff
@@ -27,105 +40,142 @@ export default function App({ loadingHtml, maxResourceSize, lfsServer, orgId, da
+            setLinkUrl(data.url);
         } else {
             // resource has no file or link
             setHiddenInputs(null, {});
-        };
+        }
         setUseEffectCompleted(true);
     }, []);
 
     if (!useEffectCompleted) {
-        return <div dangerouslySetInnerHTML={{ __html: loadingHtml }}></div>
+        return <div dangerouslySetInnerHTML={{ __html: loadingHtml }}></div>;
     }
 
-    if (uploadError) return (
-        <div className="alert alert-danger">
-            <p><i className="fa fa-exclamation-triangle"></i> {uploadError.error}</p>
-            <p>
-                <span>{uploadError.description}</span>
-                <br />
-                <span>{ckan.i18n._('Please refresh this page and try again.')}</span>
-            </p>
-        </div>
-    )
+    if (uploadError)
+        return (
+            <div className="alert alert-danger">
+                <p>
+                    <i className="fa fa-exclamation-triangle"></i>{' '}
+                    {uploadError.error}
+                </p>
+                <p>
+                    <span>{uploadError.description}</span>
+                    <br />
+                    <span>
+                        {ckan.i18n._('Please refresh this page and try again.')}
+                    </span>
+                </p>
+            </div>
+        );
 
     function UploaderComponent() {
-        const resetComponent = e => {
+        const resetComponent = (e) => {
             setHiddenInputs(null, {});
             setUploadProgress(defaultUploadProgress);
             e.preventDefault();
-        }
+        };
         if (uploadProgress.total) {
             const loaded =
-                uploadProgress.loaded == uploadProgress.total
-                && uploadfileName;
-            return (
-                loaded
-                    ? <DisplayUploadedFile {...{
+                uploadProgress.loaded == uploadProgress.total && uploadfileName;
+            return loaded ? (
+                <DisplayUploadedFile
+                    {...{
                         fileName: uploadfileName,
-                        resetComponent: resetComponent
-                    }} />
-                    : <ProgressBar {...{ uploadProgress }} />
-            )
+                        resetComponent: resetComponent,
+                    }}
+                />
+            ) : (
+                <ProgressBar {...{ uploadProgress }} />
+            );
+        }
+        if (uploadMode === 'url') {
+            return <UrlUploader {...{ linkUrl, resetComponent }} />;
+        } else if (uploadMode === 'resource') {
+            return (
+                <ResourceForker
+                    selectedResource={selectedResource}
+                    setSelectedResource={setSelectedResource}
+                    setHiddenInputs={setHiddenInputs}
+                />
+            );
+        } else {
+            // e.g. [undefined, null, "file"].includes(uploadMode)
```

Comment:
I don't understand this comment. What's it about?

Reply:
The code was originally written to do `if <that line>; else...` and I flipped it around so I kept that more explicit statement in so that future us can understand what that is actually trying to catch. Perhaps the correct solution is to make this branch an `elseif` and have an `else` that throws an exception.
(Philosophically I would actually say that the correct solution should be to use an enum but you're using JS not TS... one for a broader discussion I'm sure :-) )